src/flamenco/capture/fd_solcap_writer.c

```diff
@@ -145,551 +16,157 @@ fd_solcap_writer_footprint( void ) {
-    fd_pubkey_t const *  stake_acc_addr,
-    fd_pubkey_t const *  vote_acc_addr,
-    uint                 commission,
-    long                 vote_rewards,
-    long                 stake_rewards,
-    long                 new_credits_observed
-) {
-  fd_solcap_StakeRewardEvent event = {
-    .commission           = commission,
-    .vote_rewards         = vote_rewards,
-    .stake_rewards        = stake_rewards,
-    .new_credits_observed = new_credits_observed
-  };
-  memcpy( event.stake_account_address, stake_acc_addr, 32UL );
-  memcpy( event.vote_account_address,  vote_acc_addr,  32UL );
-  return fd_solcap_write_protobuf( writer, &event, fd_solcap_StakeRewardEvent_fields, FD_SOLCAP_V1_REWARD_CALC_MAGIC );
-}
-
-int
-fd_solcap_write_vote_account_payout(
-    fd_solcap_writer_t * writer,
-    fd_pubkey_t const *  vote_acc_addr,
-    ulong                update_slot,
-    ulong                lamports,
-    long                 lamports_delta
-) {
-  fd_solcap_VoteAccountPayout payout = {
-    .update_slot    = update_slot,
-    .lamports       = lamports,
-    .lamports_delta = lamports_delta
-  };
-  memcpy( payout.address, vote_acc_addr, 32UL );
-  return fd_solcap_write_protobuf( writer, &payout, fd_solcap_VoteAccountPayout_fields, FD_SOLCAP_V1_REWARD_VOTE_MAGIC );
-}
-
-int
-fd_solcap_write_stake_account_payout(
-    fd_solcap_writer_t * writer,
-    fd_pubkey_t const *  stake_acc_addr,
-    ulong                update_slot,
-    ulong                lamports,
-    long                 lamports_delta,
-    ulong                credits_observed,
-    long                 credits_observed_delta,
-    ulong                delegation_stake,
-    long                 delegation_stake_delta
-) {
-  fd_solcap_StakeAccountPayout payout = {
-    .update_slot            = update_slot,
-    .lamports               = lamports,
-    .lamports_delta         = lamports_delta,
-    .credits_observed       = credits_observed,
-    .credits_observed_delta = credits_observed_delta,
-    .delegation_stake       = delegation_stake,
-    .delegation_stake_delta = delegation_stake_delta
-  };
-  memcpy( payout.address, stake_acc_addr, 32UL );
-  return fd_solcap_write_protobuf( writer, &payout, fd_solcap_StakeAccountPayout_fields, FD_SOLCAP_V1_REWARD_STAKE_MAGIC );
-}
-
-int
-fd_solcap_write_protobuf( fd_solcap_writer_t *        writer,
-                          void const *                msg,
-                          struct pb_msgdesc_s const * desc,
-                          ulong                       magic ) {
-  if( FD_UNLIKELY( !writer ) ) return 0;
-
-  uchar buf[ 1UL<<20 ];
-  pb_ostream_t stream = pb_ostream_from_buffer( buf, sizeof(buf) );
-  if( FD_UNLIKELY( !pb_encode( &stream, desc, msg ) ) ) {
-    FD_LOG_WARNING(( "pb_encode failed (%s)", PB_GET_ERROR(&stream) ));
-    return EPROTO;
+  if (padding_needed > 0) {
+    static const char zeros[4] = {0};
+    FD_TEST(padding_needed == write(fd, zeros, padding_needed));
```

Comment:
Doing a syscall to write 4 padding bytes seems very inefficient, consider the use of fd_io_buffered